### DISCUSSION TITLE
unit and integration testing

### DISCUSSION THREAD

**Discussion by bahag-raesenerm** [csells/go_router#312] [Q&A]
Hey everybody.

First of all, thx to Chris for this package. It's really awesome as far as I can tell.

Playing around with it I was thinking about testability, especially regarding testability of deep linking. I expect we will go with BloC and could unit test "regular" routes quite well with this, but so far I couldn't find much informations about testing deep linking.

Any thoughts on this appreciated.

Marius

**Comment by csells** [comment]
There's an excellent article on just this topic you may find useful: https://guillaume.bernos.dev/testing-go-router/

**Reply by bahag-raesenerm** [reply]
now I wonder why I couldn't find this despite using our both, presumingly, search engine of choice - but anyways, thx for pointing to the article, I'll work through it.
